### PR TITLE
feat(rollup-verifier): codecv4

### PR DESCRIPTION
**Pull request by colinlyguo**
## 1. Purpose or design rationale of this PR

1. store `version` and `versionedHash` in the database.
2. glue code for backward compatibility, can be removed after upgrade.

## 2. PR title

Your PR title must follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) (as we are doing squash merge for each PR), so it must start with one of the following [types](https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#type):

- [x] feat: A new feature

## 3. Deployment tag versioning

Has the version in `params/version.go` been updated?

- [x] Yes

## 4. Breaking change label

Does this PR have the `breaking-change` label?

- [x] This PR is not a breaking change

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -50,7 +50,7 @@ require (
 	github.com/prometheus/tsdb v0.7.1
 	github.com/rjeczalik/notify v0.9.1
 	github.com/rs/cors v1.7.0
-	github.com/scroll-tech/da-codec v0.1.1-0.20240718144756-1875fd490923
+	github.com/scroll-tech/da-codec v0.1.1-0.20240819100936-c6af3bbe7068
```
Comment (Thegaram) [inline go.mod]: Can we merge that PR first?

File: rollup/rollup_sync_service/rollup_sync_service.go [hunk 2]
```diff
@@ -342,9 +347,9 @@ func (s *RollupSyncService) getLocalChunksForBatch(batchIndex uint64) ([]*encodi
 	return chunks, nil
 }
 
-func (s *RollupSyncService) getChunkRanges(batchIndex uint64, vLog *types.Log) ([]*rawdb.ChunkBlockRange, error) {
+func (s *RollupSyncService) getCommittedBatchMeta(batchIndex uint64, vLog *types.Log) (*rawdb.CommittedBatchMeta, []*rawdb.ChunkBlockRange, error) {
 	if batchIndex == 0 {
-		return []*rawdb.ChunkBlockRange{{StartBlockNumber: 0, EndBlockNumber: 0}}, nil
+		return nil, []*rawdb.ChunkBlockRange{{StartBlockNumber: 0, EndBlockNumber: 0}}, nil
```
Comment (Thegaram) [inline rollup/rollup_sync_service/rollup_sync_service.go]: Why use `nil` here? Won't that break anything on a new network / new node?

**Review by Thegaram — COMMENTED** [review]
LGTM

File: go.mod (hunk 1 above)
Comment (colinlyguo) [reply go.mod]: sounds good.

File: rollup/rollup_sync_service/rollup_sync_service.go (hunk 2 above)
Comment (colinlyguo) [reply rollup/rollup_sync_service/rollup_sync_service.go]: nice catch. my negligence. fixed in [15aafe1](https://github.com/scroll-tech/go-ethereum/pull/991/commits/15aafe1303265d3ee9d39e41b5be6e4263e6cc91). will run a follower node to test, too.

File: go.mod (hunk 1 above)
Comment (colinlyguo) [reply go.mod]: updated in [cffc8a1](https://github.com/scroll-tech/go-ethereum/pull/991/commits/cffc8a16819c7be7c611ed47a2428dc8377a4170).

**Review by Thegaram — DISMISSED** [review]

**Review by zimpha — DISMISSED** [review]

**Review by zimpha — APPROVED** [review]

**Review by 0xmountaintop — APPROVED** [review]

**Pull request merged**